lib/formatter.rb

```diff
@@ -1,22 +1,36 @@
 class Formatter
+  def initialize(colorizer:, symbols:)
+    @colorizer = colorizer
+    @symbols = symbols
+  end
+
   def format(board)
     flattened = flatten(board)
-    replaced = replace_nil_with_position(flattened)
-    rows = replaced.each_slice(board.size).to_a
+    colored = color_and_replace_nil(flattened)
+    rows = colored.each_slice(board.size).to_a
     rows_and_dividers(rows).join("\n")
   end
 
   private
 
+  attr_reader :colorizer, :symbols
+
   def flatten(board)
     (1..board.size**2).to_a.map do |position|
       board.get(at: position)
     end
   end
 
-  def replace_nil_with_position(flattened)
+  def color_and_replace_nil(flattened)
     flattened.map.with_index do |symbol, position|
-      symbol.nil? ? position + 1 : symbol
+      case symbol
```

Comment:
Would you want a default color? Maybe the nil case, magenta, should be default?

Reply:
That is cleaner, thanks!